pom.xml

```diff
@@ -79,7 +79,7 @@
     <!-- Hadoop and Hbase-thirdparty version -->
     <!-- These are expected to be overridden to conform to cluster versions
     along with hbase.version (defined in profiles) -->
-    <hadoop.version>3.0.3</hadoop.version>
+    <hadoop.version>3.1.4</hadoop.version>
```

Comment:
Any chance we can use Hadoop 3.2.x version's APIs? 3.1.x is not production ready [announcement](https://lists.apache.org/thread.html/8313e605c0ed0012f134cce9cc6adca738eea81feccea99c8de87cd9@%3Cgeneral.hadoop.apache.org%3E)
Although it might not impact our usecase but if users are not going to run Hadoop 3.1.x in prod, we can also use stable version as downstreamers?

Reply:
The announcement you quote refers specifcally to  3.1.0, not to later 3.1.x versions.
3.1.4 is production ready.

Having said that, I am not against going to Hadoop 3.2, but I'd prefer to leave that until after the 5.1 release, as we have very litte time to shake out any possblie problems.

HBase 2.1 doesn't officially support Hadoop 3.2 either (though it's probably more a question of the relative release times than an actual incompatibility)

Reply:
> HBase 2.1 doesn't officially support Hadoop 3.2 either (though it's probably more a question of the relative release times than an actual incompatibility)

True, as long as we are going to support HBase 2.1 with Phoenix 5.1 (which is the plan), we will have to adhere to this.

> The announcement you quote refers specifcally to 3.1.0, not to later 3.1.x versions.
3.1.4 is production ready.

Sure this sounds good.

Reply:
> > HBase 2.1 doesn't officially support Hadoop 3.2 either (though it's probably more a question of the relative release times than an actual incompatibility)
> 
> True, as long as we are going to support HBase 2.1 with Phoenix 5.1 (which is the plan), we will have to adhere to this.

We can override hbase.version from each hbase profile, so we could use 3.1.4 for 2.1, and 3.2.1 for 2.2+.

We should also consider dropping HBase 2.1 support in 5.2

Reply:
> We can override hbase.version from each hbase profile, so we could use 3.1.4 for 2.1, and 3.2.1 for 2.2+.

Actually this one is quite better choice.

> We should also consider dropping HBase 2.1 support in 5.2

Absolutely, it is EOL'ed anyways.